### DISCUSSION TITLE
Using a secondary IP address for IPSEC

### DISCUSSION THREAD

**Discussion by AS7923** [strongswan/strongswan#1084] [Q&A]
I want to use a secondary IP address for encapsulating IPSEC traffic. I cannot seem to get this working. The ip xfrm policy that strongswan creates has selectors only for the primary IP address and I can't seem to influence this. 
Is this really a restriction or am I doing something wrong ?

**Comment by tobiasbrunner** [comment]
Not sure I understand your question, but if your intention is to use a different IP address for ESP than for IKE then that's not possible.

**Reply by AS7923** [reply]
No, that is not what I want. I want the same address for both IKE and ESP. But this address should be the secondary IP address on the interface.

**Reply by tobiasbrunner** [reply]
Then configure that address in `local_addrs`. (Or create appropriate source routes that force that IP to get used.)
